tests/data-type.document.details.spec.js

```diff
@@ -0,0 +1,89 @@
+const expect = require('chai').expect;
+const { DocumentDetails } = require('../src/data_type/document.details');
+
+describe('documentDetails', () => {
+  context('when value is empty', () => {
+    it('it should throw an exception', () => {
+      try {
+        const documentDetails = new DocumentDetails('');
+      } catch (err) {
+        expect(err.message).to.equal('Invalid value for DocumentDetails');
+      }
+    });
+  });
+  context('when value is less than 3 words', () => {
+    it('it should throw an exception', () => {
+      try {
+        const documentDetails = new DocumentDetails('PASS_CARD GBR');
+      } catch (err) {
+        expect(err.message).to.equal('Invalid value for DocumentDetails');
+      }
+    });
+  });
+  context('when value is four words', () => {
+    it('it should parse one optional attribute', () => {
+      const documentDetails = new DocumentDetails('PASSPORT GBR 01234567 2020-01-01');
+      expect(documentDetails.type).to.equal('PASSPORT');
+      expect(documentDetails.issuingCountry).to.equal('GBR');
+      expect(documentDetails.documentNumber).to.equal('01234567');
+      expect(documentDetails.expirationDate.toISOString().slice(0,10)).to.equal('2020-01-01');
```

Comment:
What else is on the String if we do not _slice()_ it?

Reply:
This one of the way to get the format the date in the short format `Y-d-m` in javascript.
If you don't slice it, you will get the full date as `2020-01-01T00:00:00.000Z`

Reply:
Is there any other js _Date_ type that only applies to Dates?  I mean, without a time component?